src/navigation/image_database.cc

```diff
@@ -458,16 +458,23 @@ ImageDatabase::getFilename(const Vector3<millimeter_t> &position,
 bool
 ImageDatabase::fileNameCompare(const std::string &fn1, const std::string &fn2)
 {
+    // Find file names like (string)(number)(extension), e.g. "image45.png"
```

Comment:
What's the purpose of the group around everything? Can't that be just be accessed with ``m[0]``?

Reply:
Those brackets aren't part of the regex -- they show where the raw string begins and ends. It's a weird syntax.

Reply:
Oh yeah - **those** brackets. I'm sure there's some reason why they're necessary but I have never understood it....